yt_dlp/extractor/rokfin.py

```diff
@@ -0,0 +1,601 @@
+# coding: utf-8
+from json import (
+    dumps as json_dumps,
+)
+from math import ceil as math_ceil
+from itertools import (
+    count as itertools_count,
+    product as itertools_product,
+)
+import datetime
+from re import finditer as re_finditer
+from .common import InfoExtractor, SearchInfoExtractor
+from ..utils import (
+    unified_timestamp,
+    try_get,
+    traverse_obj,
+    int_or_none,
+    bool_or_none,
+    float_or_none,
+    str_or_none,
+    url_or_none,
+    ExtractorError,
+)
+
+
+class RokfinSingleVideoIE(InfoExtractor):
+    _META_DATA_BASE_URL = 'https://prod-api-v2.production.rokfin.com/api/v2/public/'
+    _RECOMMENDED_VIDEO_BASE_URL = 'https://rokfin.com/'
+    _CHANNEL_BASE_URL = 'https://rokfin.com/'
+
+
+class RokfinPostIE(RokfinSingleVideoIE):
+    IE_NAME = 'rokfin:post'
+    _VALID_URL = r'https?://(?:www\.)?rokfin\.com/(?P<id>post/[^/]+)'
+    _TESTS = [{
+        'url': 'https://www.rokfin.com/post/57548/Mitt-Romneys-Crazy-Solution-To-Climate-Change',
+        'info_dict': {
+            'id': 'post/57548',
+            'ext': 'mp4',
+            'title': 'Mitt Romney\'s Crazy Solution To Climate Change',
+            'thumbnail': 're:https://img.production.rokfin.com/.+',
+            'upload_date': '20211023',
+            'timestamp': 1634998029,
+            'creator': 'Jimmy Dore',
+            'channel': 'Jimmy Dore',
+            'channel_id': 65429,
+            'channel_url': 'https://rokfin.com/TheJimmyDoreShow',
+            'duration': 213.0,
+            'availability': 'public',
+            'is_live': False,
+            'was_live': False,
+            'live_status': 'not_live'
+        }
+    }]
+
+    def _real_extract(self, url_from_user):
+        video_id = self._match_id(url_from_user)
+        downloaded_json = self._download_json(self._META_DATA_BASE_URL + video_id, video_id, note='Downloading video metadata', fatal=False) or {}
```

Comment:
```suggestion
        downloaded_json = self._download_json(self._META_DATA_BASE_URL + video_id, video_id)
```
everything (especially formats) depends on this dict so it must be fatal

Reply:
forgor to mention
change the name for this dict
`video_json` or `data_json` are better

Reply:
> everything (especially formats) depends on this dict so it must be fatal

No. With fatal=True, downloading a non-existing video, for example, results in an error message saying "please report this issue". This indicates that there is a bug. Sure, the error message may be suppressed by caching HTTPError and other exceptions, as well, and re-throwing them with a custom message, but that's more complex than it's worth.